vaccine_feed_ingest/apis/placekey.py

```diff
@@ -0,0 +1,77 @@
+from typing import Optional
+
+import diskcache
+import placekey.api
+
+from ..utils.log import getLogger
+from .common import CachedAPI, calculate_cache_key
+
+logger = getLogger(__file__)
+
+
+class PlacekeyAPI(CachedAPI):
+    """API for calling placekey that checks the cache first"""
+
+    def __init__(self, api_cache: diskcache.Cache, apikey: str):
+        self._placekey_api = placekey.api.PlacekeyAPI(apikey)
+        super().__init__(api_cache)
+
+    def lookup_placekey(
+        self,
+        latitude: float,
+        longitude: float,
+        location_name: str,
+        street_address: str,
+        city: str,
+        region: str,
+        postal_code: str,
+        iso_country_code: str = "US",
+        strict_address_match: bool = False,
+        strict_name_match: bool = False,
+    ) -> Optional[str]:
+        cache_key = calculate_cache_key(
+            "placekey",
+            [
+                f"{latitude:.5f}",
+                f"{longitude:.5f}",
+                location_name,
+                street_address,
+                city,
+                region,
+                postal_code,
+                iso_country_code,
+                str(strict_address_match),
+                str(strict_name_match),
+            ],
+        )
+
+        cache_response = self.api_cache.get(cache_key)
+        if cache_response:
+            return cache_response.get("placekey")
```

Comment:
Should we check if the cached value was an error (and return `None` if so)?

Reply:
The `.get` will return `None` if there was a cached `error`

Reply:
I will explicitly return `None` with there is an error though